app/src/main/java/com/duckduckgo/app/widget/AddWidgetLauncher.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.widget
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.app.ActivityOptions
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import com.duckduckgo.app.widget.ui.AddWidgetInstructionsActivity
+import com.duckduckgo.app.widget.ui.WidgetCapabilities
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.widget.SearchAndFavoritesWidget
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+interface AddWidgetLauncher {
+    fun launchAddWidget(activity: Activity?)
+}
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class AddWidgetCompatLauncher @Inject constructor(
+    private val defaultAddWidgetLauncher: AppWidgetManagerAddWidgetLauncher,
```

Comment:
If both,`AppWidgetManagerAddWidgetLauncher` and `LegacyAddWidgetLauncher` are concrete classes of `AddWidgetLauncher`, we should inject them as `AddWidgetLauncher`. Otherwise, there's no benefit of them implementing the same interface.